README.md

```diff
@@ -4,8 +4,164 @@ This is a [Singer](https://singer.io) tap that produces JSON-formatted data
+    OR
+    > cd .../tap-pinterest-ads
+    > pip install .
+    ```
+2. Dependent libraries
+    The following dependent libraries were installed.
+    ```bash
+    > pip install singer-python
+    > pip install singer-tools
+    > pip install target-stitch
+    > pip install target-json
+    
+    ```
+    - [singer-tools](https://github.com/singer-io/singer-tools)
+    - [target-stitch](https://github.com/singer-io/target-stitch)
+
+3. Here is what the config should look like
+
+    ```json
+    {
+        "start_date": "2019-01-01T00:00:00Z",
+        "client_id": <CLIENT_ID>,
+        "client_secret": <CLIENT_SECRET>,
+        "redirect_uri": <REDIRECT_URI>,
+        "refresh_token": <REFRESH_TOKEN>,
+        "quota_limit": <QUOTA_LIMIT>
+    }
+    ```
+    The following are required values: `start_date`, `client_id`, `client_secret`, `redirect_uri`, `refresh_token`
+    The following are optional values: `quota_limit`
+
+    Optionally, also create a `state.json` file. `currently_syncing` is an optional attribute used for identifying the last object to be synced in case the job is interrupted mid-stream. The next run would begin where the last job left off.
+
+    ```json
+      {
+        "bookmarks": {
+          "stream_1": "2020-01-04T00:00:00.000000Z",
+          "stream_2": 1576266090,
+          ...
+        }
+      }
+    ```
+
+4. Run the Tap in Discovery Mode
+    This creates a catalog.json for selecting objects/fields to integrate:
+    ```bash
+    tap-outreach --config config.json --discover > catalog.json
+    ```
+   See the Singer docs on discovery mode
+   [here](https://github.com/singer-io/getting-started/blob/master/docs/DISCOVERY_MODE.md#discovery-mode).
+
+5. Run the Tap in Sync Mode (with catalog) and [write out to state file](https://github.com/singer-io/getting-started/blob/master/docs/RUNNING_AND_DEVELOPING.md#running-a-singer-tap-with-a-singer-target)
+
+    For Sync mode:
+    ```bash
+    > tap-outreach --config tap_config.json --catalog catalog.json > state.json
+    > tail -1 state.json > state.json.tmp && mv state.json.tmp state.json
+    ```
+    To load to json files to verify outputs:
+    ```bash
+    > tap-outreach --config tap_config.json --catalog catalog.json | target-json > state.json
+    > tail -1 state.json > state.json.tmp && mv state.json.tmp state.json
+    ```
+    To pseudo-load to [Stitch Import API](https://github.com/singer-io/target-stitch) with dry run:
+    ```bash
+    > tap-outreach --config tap_config.json --catalog catalog.json | target-stitch --config target_config.json --dry-run > state.json
+    > tail -1 state.json > state.json.tmp && mv state.json.tmp state.json
+    ```
+
+6. Test the Tap
+    
+    While developing the PINTEREST-ADS tap, the following utilities were run in accordance with Singer.io best practices:
+    Pylint to improve [code quality](https://github.com/singer-io/getting-started/blob/master/docs/BEST_PRACTICES.md#code-quality):
```

Comment:
If we're recommending people use `pylint` should we tell them the checks we tend to disable?

Reply:
yeah, where is that list?

Reply:
`pylint <tap_name> -d broad-except,chained-comparison,empty-docstring,fixme,invalid-name,line-too-long,missing-class-docstring,missing-function-docstring,missing-module-docstring,no-else-raise,no-else-return,too-few-public-methods,too-many-arguments,too-many-branches,too-many-lines,too-many-locals,ungrouped-imports,wrong-spelling-in-comment,wrong-spelling-in-docstring`